src/utils/register-controllers.ts

```diff
@@ -9,10 +12,13 @@ export function registerControllers(
   classes: ObjectKeyAny[] = [],
   addToRoute: (route: RouteMetadata) => void,
   logging: boolean = true,
+  customContainer?: DependencyContainer,
 ) {
   // TODO: add two route Map (with route params / exact match)
   // example: new Map(); key = route, value = object
 
+  const container = customContainer || defaultContainer;
```

Comment:
If we want to use customContainer everywhere, it must still be in the UseHook decorator.

Reply:
How does UseHook work? All I can find is this https://github.com/alosaur/alosaur/blob/master/src/decorator/UseHook.ts and I'm not really that familiar with making ts decorators

Reply:
you need to add appSettings to MetadataArgsStorage and take this custom container from there.

let me know if there are any problems with this, I will accept PR without it, then add support to UseHook